packages/astro/components/Code.astro

```diff
@@ -0,0 +1,21 @@
+---
+import shiki from 'shiki';
+const { code, lang = 'plaintext', theme = 'nord' } = Astro.props;
+
+/** Replace the shiki class name with a custom astro class name. */
+function repairClassName(html: string): string {
+  html = html.replace('<pre class="shiki"', '<pre class="astro-code"');
+  return html;
+}
+
+/** Replace the shiki custom css theme colors with Astro-specific css variables. */
+function repairCssThemeColors(html: string): string {
+  html = html.replace(/style="(background-)?color: var\(--shiki-/g, 'style="$1color: var(--astro-code-');
+  return html;
+}
+
+const highlighter = await shiki.getHighlighter({theme});
+const _html = highlighter.codeToHtml(code, lang);
+html = repairCssThemeColors(repairClassName(_html));
+---
+{html}
```

Comment:
Just a question, is this adding a `<pre><code>` or are you expected to add those yourself?

Reply:
Shiki should be generating that as part of its themed html